modules/build/src/main/scala/scala/build/preprocessing/JavaPreprocessor.scala

```diff
@@ -109,6 +109,21 @@ final case class JavaPreprocessor(
         }
         Some(res)
 
+      case jar: JarFile => Some(either {
```

Comment:
NIT I wonder if it wouldn't be easier to follow this code path if Jars had their own preprocessor.
We probably should support virtual jars in the future, too, so there'd be more dedicated logic, really.